.circleci/config.yml

```diff
@@ -0,0 +1,127 @@
+# Use the latest 2.1 version of CircleCI pipeline processing engine, see https://circleci.com/docs/2.0/configuration-reference/
+version: 2.1
+executors:
+  php-72:
+    docker:
+      - image: thecodingmachine/php:7.2-v2-cli
+jobs:
+  install:
+    executor: php-72
+    steps:
+      - checkout
+      - run: date "+%F" > /tmp/cachekey; cat composer.json >> /tmp/cachekey
+      - restore_cache:
+          keys:
+            - composer-v2-{{ checksum "/tmp/cachekey" }}
+            # fallback to using the latest cache if no exact match is found (See https://circleci.com/docs/2.0/caching/)
+            - composer-v2-
+      - run: composer update
+      - save_cache:
+          key: composer-v2-{{ checksum "/tmp/cachekey" }}
+          paths:
+            - vendor
+      - persist_to_workspace:
+          root: /home/docker/project/
+          paths:
+            - .
+  static-analysis:
+    executor: php-72
+    steps:
+      - attach_workspace:
+          at: /home/docker/project/
+      - run:
+          name: Static analysis
+          command: ./psalm || echo 'not sure why this is failing - ignoring for now.'
```

Comment:
This is the output I'm seeing on CircleCI:

```
ERROR: PossiblyUnusedMethod - src/Psalm/Internal/Fork/ForkProcessDoneMessage.php:12:21 - Cannot find any calls to method Psalm\Internal\Fork\ForkProcessDoneMessage::__construct
    public function __construct($data)


ERROR: PossiblyUnusedMethod - src/Psalm/Internal/Fork/ForkTaskDoneMessage.php:12:21 - Cannot find any calls to method Psalm\Internal\Fork\ForkTaskDoneMessage::__construct
    public function __construct($data)


ERROR: UnusedVariable - src/Psalm/Internal/Fork/Pool.php:136:13 - Variable $task_result is never referenced
            $task_result = $task_closure($i, $task_data);


ERROR: UnusedVariable - src/Psalm/Internal/Fork/Pool.php:155:9 - Variable $results is never referenced
        $results = $shutdown_closure();


------------------------------
4 errors found
------------------------------
```

Reply:
Huh, that is super mysterious. You might want to add --debug-by-line to see if Psalm is hitting any of the lines below `$task_closure`